sphinxcontrib/openapi/openapi31.py

```diff
@@ -473,11 +473,15 @@ def openapihttpdomain(spec, **options):
     # Remove paths matching regexp
     if "exclude" in options:
         _paths = []
-        for e in options["exclude"]:
-            er = re.compile(e)
-            for path in paths:
-                if not er.match(path):
-                    _paths.append(path)
+        compiled_options = [re.compile(e) for e in options["exclude"]]
+        for path in paths:
+            matched = False
+            for er in compiled_options:
+                if er.match(path):
+                    matched = True
+                    break
+            if not matched:
+                _paths.append(path)
```

Comment:
Good catch. I wonder if we could simplify somewhat though? Something like:

```
if not any(re.match(pattern, path) for pattern in options["exclude"]):
    _paths.append(path)
```

(the [`re` module caches compiled patterns](https://bugs.python.org/issue17441) so there's no real advantage to calling `re.compile` ahead of time)

Reply:
Ah that's great, updated with simplified code